Core/TopBuilder.cpp

```diff
@@ -285,14 +285,17 @@ void Builder::accessChainStore(llvm::Value* value)
 {
     assert(accessChain.isRValue == false);
 
-    llvm::Value* base = collapseAccessChain();
-    llvm::Value* source = value;
-
-    // if swizzle exists, it is out-of-order or not full, we must load the target vector,
-    // extract and insert elements to perform writeMask and/or swizzle
-    if (accessChain.swizzle.size()) {
-
-        llvm::Value* shadowVector = createLoad(base, accessChain.metadataKind, accessChain.mdNode);
+    // if swizzle exists, it is out-of-order or not full, we must
+    // extract and store each element to perform writeMask and/or swizzle
+    if (accessChain.swizzle.size() > 1) {
+
+        llvm::Value* base = collapseAccessChain();
```

Comment:
This will still be missing the component selection.  Probably better to add the component to the chain, and then collapse it.